File: Data Structures & Algorithms/balanced-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isBalanced(TreeNode* root) {
        if(root==nullptr)
            return true; 
        if(abs(height(root->left)-height(root->right))>1 )
            return false; 
        return isBalanced(root->left) && isBalanced(root->right);

    }
    int height(TreeNode* root){
        if(root==nullptr)
            return 0;
        int left= 1+height(root->left);
        int right=1+height(root->right);
        return max(left,right);

    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        result = [] 
        if not root:
            return 0
        stack = deque() #assuming the root is the highest value. 
        stack.append((root,root.val))
        result = []
        result.append(root.val)
        while stack:
            element,maxi = stack.pop()
            if element.left: 
                stack.append((element.left,max(element.left.val,maxi)))
                if element.left.val>=maxi: 
                    result.append(element.left.val)
            if element.right: 
                stack.append((element.right,max(element.right.val,maxi)))  
                if element.right.val>=maxi: 
                    result.append(element.right.val) 

        return len(result) 
